### PR TITLE
Dra 1579 dra 1580 holdback tabeller

### PR DESCRIPTION
**Pull request by davidgrove73**

### CODE REVIEW

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
I believe these tables should be created with their contents. So the DDL files should contain information to populate the tables, so that they represent the values from the old excel sheets.

**Comment by thomasegense** [comment]
Make the  initial insert data in another file (.SQL) so seperate it from the .ddl file.

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.04', 'Undervisning', 2190);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.05', 'Dramatik & Fiktion', 3650);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.06', 'Musik', 2190);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.07', 'Underholdning', 2190);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.08', 'Sport', 2190);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.09', 'Præsentation', 2190);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.09*', 'Øvrigt (TTV + Pauser)', 2190);
+INSERT INTO DR_HOLDBACK_RULES (id, name, days) VALUES ('2.20', 'Sponsorskilte', 2190);
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (1, INT4RANGE(1000, 1900), INT4RANGE(1000, 1000), '2.02');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:14]: All ranges should include the edge values. I believe a line like this should be written like this to include the last value in the range: 
`INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (1, INT4RANGE(1000, 1900, '[]'), INT4RANGE(1000, 1000, '[]'), '2.02');`

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (10, INT4RANGE(2000, 2900), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (11, INT4RANGE(2000, 2900), INT4RANGE(1200, 1500), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (12, INT4RANGE(2000, 2900), INT4RANGE(1600, 1600), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (13, INT4RANGE(2000, 2900), INT4RANGE(1700, 1700), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (14, INT4RANGE(2000, 2900), INT4RANGE(1800, 1800), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (15, INT4RANGE(2000, 2900), INT4RANGE(1900, 1900), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (16, INT4RANGE(2000, 2900), INT4RANGE(2000, 2300), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (17, INT4RANGE(2000, 2900), INT4RANGE(4000, 4900), '2.03');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:31]: There should be 10 entries for this line in the excel sheet

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (26, INT4RANGE(3200, 3390), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (27, INT4RANGE(3200, 3390), INT4RANGE(1200, 1500), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (28, INT4RANGE(3200, 3390), INT4RANGE(1600, 1600), '2.06');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (29, INT4RANGE(3200, 3390), INT4RANGE(1700, 1700), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (30, INT4RANGE(3200, 3390), INT4RANGE(1800, 1800), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (31, INT4RANGE(3200, 3390), INT4RANGE(1900, 1900), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (32, INT4RANGE(3200, 3390), INT4RANGE(2000, 2300), '2.06');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (33, INT4RANGE(3200, 3390), INT4RANGE(3000, 3900), '2.06');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:49]: A line is missing here as well, might be FORM 9, 10 or 11

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (34, INT4RANGE(4000, 4900), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (35, INT4RANGE(4000, 4900), INT4RANGE(1200, 1500), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (36, INT4RANGE(4000, 4900), INT4RANGE(1600, 1600), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (37, INT4RANGE(4000, 4900), INT4RANGE(1700, 1700), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (38, INT4RANGE(4000, 4900), INT4RANGE(1800, 1800), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (39, INT4RANGE(4000, 4900), INT4RANGE(1900, 1900), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (40, INT4RANGE(4000, 4900), INT4RANGE(2000, 2300), '2.03');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:57]: One form value is missing here

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (42, INT4RANGE(5000, 5990), INT4RANGE(1600, 1600), '2.08');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (43, INT4RANGE(5000, 5990), INT4RANGE(2000, 2300), '2.08');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (44, INT4RANGE(5000, 5990), INT4RANGE(3000, 3900), '2.08');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (45, INT4RANGE(5000, 5990), INT4RANGE(4000, 4900), '2.08');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (46, INT4RANGE(5000, 5990), INT4RANGE(5000, 5900), '2.08');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (47, INT4RANGE(5000, 5990), INT4RANGE(6000, 6000), '2.08');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (48, INT4RANGE(5000, 5990), INT4RANGE(6100, 6100), '2.01.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (49, INT4RANGE(5000, 5990), INT4RANGE(6200, 6900), '2.08');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:67]: Are there to many entries here, I only see entries for form 4, 9, 12, 13 and 14 in the sheet

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (50, INT4RANGE(6000, 6300), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (51, INT4RANGE(6000, 6300), INT4RANGE(1200, 1500), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (52, INT4RANGE(6000, 6300), INT4RANGE(1600, 1600), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (53, INT4RANGE(6000, 6300), INT4RANGE(1700, 1700), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (54, INT4RANGE(6000, 6300), INT4RANGE(1800, 1800), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (55, INT4RANGE(6000, 6300), INT4RANGE(1900, 1900), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (56, INT4RANGE(6000, 6300), INT4RANGE(2000, 2300), '2.07');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:75]: Missing form 10 in this row

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (57, INT4RANGE(6400, 6400), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (58, INT4RANGE(6400, 6400), INT4RANGE(1200, 1500), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (59, INT4RANGE(6400, 6400), INT4RANGE(1600, 1600), '2.05');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (60, INT4RANGE(6400, 6400), INT4RANGE(1700, 1700), '2.05');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (61, INT4RANGE(6400, 6400), INT4RANGE(1800, 1800), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (62, INT4RANGE(6400, 6400), INT4RANGE(1900, 1900), '2.07');
+
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:83]: Missing three form values in this row

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (63, INT4RANGE(6500, 6600), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (64, INT4RANGE(6500, 6600), INT4RANGE(1200, 1500), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (65, INT4RANGE(6500, 6600), INT4RANGE(1600, 1600), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (66, INT4RANGE(6500, 6600), INT4RANGE(1700, 1700), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (67, INT4RANGE(6500, 6600), INT4RANGE(1800, 1800), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (68, INT4RANGE(6500, 6600), INT4RANGE(1900, 1900), '2.07');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:89]: Missing two form values in this row

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (69, INT4RANGE(6700, 6700), INT4RANGE(1000, 1000), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (70, INT4RANGE(6700, 6700), INT4RANGE(1200, 1500), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (71, INT4RANGE(6700, 6700), INT4RANGE(1600, 1600), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (72, INT4RANGE(6700, 6700), INT4RANGE(1700, 1700), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (73, INT4RANGE(6700, 6700), INT4RANGE(1800, 1800), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (74, INT4RANGE(6700, 6700), INT4RANGE(1900, 1900), '2.07');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:96]: Missing two form values here, eventhough the result is the same for all

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (75, INT4RANGE(6800, 6900), INT4RANGE(1000, 1000), '2.03');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (76, INT4RANGE(6800, 6900), INT4RANGE(1200, 1500), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (77, INT4RANGE(6800, 6900), INT4RANGE(1600, 1600), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (78, INT4RANGE(6800, 6900), INT4RANGE(1700, 1700), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (79, INT4RANGE(6800, 6900), INT4RANGE(1800, 1800), '2.07');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (80, INT4RANGE(6800, 6900), INT4RANGE(1900, 1900), '2.07');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:103]: Missing two form values

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (121, INT4RANGE(7140, 7150), INT4RANGE(4000, 4900), '2.09*');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (122, INT4RANGE(7140, 7150), INT4RANGE(5000, 5900), '2.09*');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (123, INT4RANGE(7140, 7150), INT4RANGE(6000, 6000), '2.09*');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (124, INT4RANGE(7140, 7150), INT4RANGE(6100, 6100), '2.09*');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (125, INT4RANGE(7140, 7150), INT4RANGE(6200, 6900), '2.09*');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (126, INT4RANGE(7140, 7150), INT4RANGE(7000, 7000), '2.09*');
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (127, INT4RANGE(7160, 7212), INT4RANGE(1100, 1100), '2.09');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (128, INT4RANGE(7160, 7212), INT4RANGE(1800, 1800), '2.09');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:156]: Missing a single form entry here

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (129, INT4RANGE(7220, 7220), INT4RANGE(1000, 1000), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (130, INT4RANGE(7220, 7220), INT4RANGE(1100, 1100), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (131, INT4RANGE(7220, 7220), INT4RANGE(1200, 1500), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (132, INT4RANGE(7220, 7220), INT4RANGE(1600, 1600), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (133, INT4RANGE(7220, 7220), INT4RANGE(1700, 1700), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (134, INT4RANGE(7220, 7220), INT4RANGE(1800, 1800), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (135, INT4RANGE(7220, 7220), INT4RANGE(1900, 1900), '2.02');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (136, INT4RANGE(7220, 7220), INT4RANGE(2000, 2300), '2.02');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:165]: Missing 7 form entries here

File: src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (152, INT4RANGE(7260, 7260), INT4RANGE(7000, 7000), '2.09*');
+
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (153, INT4RANGE(7270, 7270), INT4RANGE(1000, 1000), '2.20');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (154, INT4RANGE(7270, 7270), INT4RANGE(1100, 1100), '2.20');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (155, INT4RANGE(7270, 7270), INT4RANGE(1200, 1500), '2.20');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (156, INT4RANGE(7270, 7270), INT4RANGE(1600, 1600), '2.20');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (157, INT4RANGE(7270, 7270), INT4RANGE(1700, 1700), '2.20');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (158, INT4RANGE(7270, 7270), INT4RANGE(1800, 1800), '2.20');
+INSERT INTO DR_HOLDBACK_MAP(id, content_range, form_range, dr_holdback_id) VALUES (159, INT4RANGE(7270, 7270), INT4RANGE(1900, 1900), '2.20');
```
Comment (VictorHarbo) [inline src/test/resources/ddl/rightsmodule_default_holdbackdata.ddl:191]: Missing 6 form entries here

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
A comment on how we construct the ranges and then I believe I found some missing values. I'll have another look on the values after this one has been fixed

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
Two things: 
1. `Undervisning` is stil missing from the `DR_HOLDBACK_RULES` table
2. the file `rightsmodule_default_holdbackdata.ddl` should probably be a .sql file as it only performs data operations and doesn't define the structure of the table

**Review by VictorHarbo — APPROVED** [review]
Looks very good now!
